src/main/java/core/basesyntax/model/lib/Injector.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.model.lib;
+
+import core.basesyntax.model.dao.BetDao;
+import core.basesyntax.model.dao.UserDao;
+import core.basesyntax.model.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoDaoAnnotationException {
+
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            field.setAccessible(true);
+            if (field.getAnnotation(Inject.class) != null) {
+                if (checkBetDaoType(field)) {
+                    field.set(instance, Factory.getBetDao());
+                }
+                if (checkUserDaoType(field)) {
+                    field.set(instance, Factory.getUserDao());
+                }
```

Comment:
let's refactor lines 24-29

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception

Reply:
done

Reply:
done